app/models/account.rb

```diff
@@ -185,8 +185,12 @@ def subscribed?
     !subscription_expires_at.blank?
   end
 
-  def followers_domains
-    followers.reorder(nil).pluck('distinct accounts.domain')
+  # Domains potentially used for PuSH callback URLs of followers' instances
+  def followers_subscription_domains
+    domains = followers.reorder(nil).pluck('distinct accounts.domain')
+    remote_urls = followers.reorder(nil).pluck('distinct accounts.remote_url')
```

Comment:
Ah, yeah, I just copied it, but the “distinct” is useless indeed. I'm not sure how to optimize that. We could just .uniq the array, but I'm not sure it will be very useful. We could also add a “remote_url_domain” field to Account, to parse the URL only when refreshing the remote account details, and not at each private/direct message, but I'm not sure how much better that is.
One part of the change was also to compute this array only when pushing private/direct messages, and not public/unlisted ones.